browser/ntp_sponsored_images/view_counter_service_factory.cc

```diff
@@ -48,24 +48,27 @@ ViewCounterServiceFactory::~ViewCounterServiceFactory() {}
 
 KeyedService* ViewCounterServiceFactory::BuildServiceInstanceFor(
     content::BrowserContext* browser_context) const {
-  Profile* profile = Profile::FromBrowserContext(browser_context);
+  // Only NTP in normal profile uses sponsored services.
+  if (browser_context->IsOffTheRecord())
+    return nullptr;
 
-  bool is_supported_locale = false;
-  auto* ads_service = brave_ads::AdsServiceFactory::GetForProfile(profile);
-  if (!ads_service) {
-    LOG(ERROR) << "Ads service was disabled at build time!";
-  } else {
-    is_supported_locale = ads_service->IsSupportedLocale();
-  }
+  if (auto* service = g_brave_browser_process->ntp_sponsored_images_service()) {
+    Profile* profile = Profile::FromBrowserContext(browser_context);
+    bool is_supported_locale = false;
```

Comment:
This is fine to keep here for this PR, but we'll have to remove this for Super Referrer so that the decision about whether to be SI or SR is moved to somewhere inside ViewCounterService